app/javascript/components/Results/ResultGrid/ResultGrid.tsx

```diff
@@ -55,7 +68,10 @@ export const ResultGrid = ({ result }: ResultProps) => {
           {result.publishedAt   && <span className='published-date'>{result.publishedAt}</span>}
           <div className='result-title'>
             <h2 className='result-title-label'>
-              <a href={result.url} className='result-title-link'>
+              <a href={result.url}
+                className='result-title-link'
+                data-click={clickData(affiliate, module, query, position, result.url, vertical)}
+                onClick={(event) => clickTracking((event.target as HTMLLinkElement).getAttribute('data-click') || '{}')}>
```

Comment:
That's how I had it in my initial implementation, but changed it here for a few reasons:

1. I found it handy to be able to inspect the clickdata in the page itself.  And I *suspect* it might be handy for the admin team as well as they _may_ be used to being able to do so as the current SERP implementation sets the click data via a data attribute.  It might be worthwhile for me to check that assumption.
2. It made testing the module assignment for blended results done [here](https://github.com/GSA/search-gov/pull/1488/files#diff-1cb55c059efac4ba4ed9b9695bbb241dee5a3885a9ff1af03bb0b47f66189f96R49-R55) much simpler: https://github.com/GSA/search-gov/pull/1488/files#diff-23b312b55c739e2aff9ccf80cad2b6918c634292cffafd97fffd97461aef345bR26-R31  That set, dynamic setting of module code is ONLY relevant within this module and only in the blended vertical.  This could likely be tested within the onclick itself, but would probably be beyond my capacity.

I think question 1 is to figure out whether having that visible data attribute is of use to anyone, so I'll ask.

Reply:
Alright, @stevenbarragan , @yogesh27 I've done what Steven suggested and passed the data directly to `clickTracking`.  This works, and codecoverage is happy, but it feels wrong to have tests entirely devoid of expectations here: https://github.com/GSA/search-gov/blob/3b123a22a4c8969f36aed869d30589bf996909e1/app/javascript/test/ResultGrid.test.tsx

Does this seem alright?

Reply:
Hi @lorawoodford, good point you raised - I think what we could do is to check if the clickTracking method was called when the anchor tag is clicked like:
```
it("should call clickTracking when link is clicked", () => {
  const myClickTracking = jest.fn();
  const { getByText } = render(
    <YourComponent clickTracking={myClickTracking} />
  );
  const link = getByText("test result 1");
  fireEvent.click(link);
  expect(clickTracking).toHaveBeenCalledTimes(1);
});
```
but this would need a major code change across all SERP to refractor the code for this, and also similar test would be needed while implementing clickTracking for mobile view as we will be making the whole result wrapper to be clickable. I think this implementation can be in its own JIRA ticket.

@stevenbarragan your thoughts on this?

Reply:
You're right @lorawoodford. This tests are not testing!

Can we somehow do something like?

```js
fireEvent.click(title);
expect(fetch).toHaveBeenCalledWith(...);
expect(fetch).toHaveBeenCalledTimes(1);
```

I like your suggestion @yogesh27 as well.

Reply:
I wasn't very clear. I'm suggesting we can use the same test and validate the call to fetch. e.g.

```js
  global.fetch = jest.fn(() =>
    Promise.resolve({
      json: () => Promise.resolve({...}),
    })
  );

  beforeEach(() => {
    fetch.mockClear();
  });

  it('sets correct Bing click data', () => {
    render(<ResultGrid result={result} affiliate='bing_affiliate' query='query' position={1} vertical='web' />);

    const title = screen.getByText(/test result 1/i);
    fireEvent.click(title);

    expect(fetch).toHaveBeenCalledWith(...);
    expect(fetch).toHaveBeenCalledTimes(1);
  });
```

Reply:
@stevenbarragan Alright, I think that got me where I needed to be.  Tests updated in latest commit.